nion/swift/ExportDialog.py

```diff
@@ -221,56 +226,155 @@ def cancel(self) -> bool:
+                                    width=int(size.width * PIXELS_PER_INCH / CM_PER_INCH))
+        return Geometry.IntSize(height=int(round(size.height)), width=int(round(size.height)))  # Pixels
+
+    def convert_from_pixels(self, size: Geometry.IntSize, unit_type: UnitType) -> Geometry.FloatSize:
+        if size.height is None or size.width is None:
+            return Geometry.FloatSize(0.0, 0.0)
+        if unit_type == UnitType.INCHES:
+            return Geometry.FloatSize(height=size.height / PIXELS_PER_INCH, width=size.width / PIXELS_PER_INCH)
+        elif unit_type == UnitType.CENTIMETERS:
+            return Geometry.FloatSize(height=size.height * CM_PER_INCH / PIXELS_PER_INCH,
+                                      width=size.width * CM_PER_INCH / PIXELS_PER_INCH)
+        return Geometry.FloatSize(width=size.width, height=size.height)  # Pixels
 
     def close(self) -> None:
         pass
 
 
 class ExportSVGDialog:
-
-    def __init__(self, document_controller: DocumentController.DocumentController, display_item: DisplayItem.DisplayItem) -> None:
+    def __init__(self, document_controller: DocumentController.DocumentController,
+                 display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-
         self.__document_controller = document_controller
-
         u = Declarative.DeclarativeUI()
-
         if display_item.display_data_shape and len(display_item.display_data_shape) == 2:
-            display_size = Geometry.IntSize(height=4, width=4)
+            display_size = Geometry.IntSize(height=display_item.display_data_shape[0],
+                                            width=display_item.display_data_shape[1])
         else:
-            display_size = Geometry.IntSize(height=3, width=4)
-
+            display_size = Geometry.IntSize(height=3, width=5)
         handler = ExportSVGHandler(display_size)
 
         def ok_clicked() -> bool:
-            dpi = 96
-            width_px = (handler.width_model.value or display_size.width) * dpi
-            height_px = (handler.height_model.value or display_size.height) * dpi
-
+            display_shape = handler.convert_to_pixels(
+                Geometry.FloatSize(
+                    height=float(handler.height_model.value or 0),
+                    width=float(handler.width_model.value or 0)
+                ),
+                handler.previous_unit_index
+            )
             ui = document_controller.ui
             filter = "SVG File (*.svg);;All Files (*.*)"
             export_dir = ui.get_persistent_string("export_directory", ui.get_document_location())
             export_dir = os.path.join(export_dir, display_item.displayed_title)
-            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"), export_dir, filter, None)
+            path, selected_filter, selected_directory = document_controller.get_save_file_path(_("Export File"),export_dir, filter, None)
```

Comment:
Still an unnecessary formatting change, now missing a space that was there before.